src/Microsoft.Extensions.Localization.Abstractions/LocalizedString.cs

```diff
@@ -27,14 +26,24 @@ public LocalizedString([NotNull] string name, [NotNull] string value)
         /// <param name="name">The name of the string in the resource it was loaded from.</param>
         /// <param name="value">The actual string.</param>
         /// <param name="resourceNotFound">Whether the string was found in a resource. Set this to <c>false</c> to indicate an alternate string value was used.</param>
-        public LocalizedString([NotNull] string name, [NotNull] string value, bool resourceNotFound)
+        public LocalizedString(string name, string value, bool resourceNotFound)
         {
+            if (name == null)
```

Comment:
I don't plan on changing the semantics of the current code. `NotNull` does not imply null or empty. If that was the intent, the original code would've reflected this behavior.